Insights/InsightsTests/InsightsTests.swift

```diff
@@ -2,35 +2,89 @@
 //  InsightsTests.swift
 //  InsightsTests
 //
-//  Created by Robert Mogos on 04/05/2018.
 //  Copyright © 2018 Algolia. All rights reserved.
 //
 
 import XCTest
 @testable import Insights
 
 class InsightsTests: XCTestCase {
-    
-    override func setUp() {
-        super.setUp()
-        // Put setup code here. This method is called before the invocation of each test method in the class.
+  
+  override func setUp() {
+    super.setUp()
+  }
+  
+  override func tearDown() {
+    super.tearDown()
+  }
+  
+  func testInitShouldFail() {
+    do {
+      _ = try Insights.shared(index: "test")
+    } catch let err {
+      XCTAssertNotNil(err)
     }
+  }
+  
+  func testInitShouldWork() {
+    do {
+      let indexName = "testIndex"
+      
+      let insightsRegister = Insights.register(appId: "testApp", apiKey: "testKey", indexName: indexName)
+      XCTAssertNotNil(insightsRegister)
+      
+      let insightsShared = try Insights.shared(index: indexName)
+      XCTAssertNotNil(insightsShared)
+      
+      XCTAssertEqual(insightsRegister, insightsShared, "Getting the Insights instance from register and shared should be the same")
+    } catch _ {}
+  }
+  
+  func testEventIsSentCorrectly() {
+    let expectation = self.expectation(description: "Wait for nothing")
+    let indexName = "testIndex"
+    let clickData: [String : Any] = [
```

Comment:
Took me some time to understand what this variable is used for. Can we make it a `var expectedData` here?

Reply:
OK